prov/efa/src/rxr/rxr_pkt_type_misc.c

```diff
@@ -63,45 +63,58 @@ ssize_t rxr_pkt_init_handshake(struct rxr_ep *ep,
 
 	pkt_entry->pkt_size = sizeof(struct rxr_handshake_hdr)
 			      + RXR_NUM_PROTOCOL_VERSION * sizeof(uint64_t);
-	pkt_entry->addr = addr;
+	pkt_entry->addr = tx_entry->addr;
+	pkt_entry->x_entry = tx_entry;
 	return 0;
 }
 
-void rxr_pkt_post_handshake(struct rxr_ep *ep,
-			    struct rxr_peer *peer,
-			    fi_addr_t addr)
+ssize_t rxr_pkt_post_handshake(struct rxr_ep *ep, struct rxr_peer *peer,
+				fi_addr_t addr)
 {
-	struct rxr_pkt_entry *pkt_entry;
+	struct rxr_tx_entry *tx_entry;
```

Comment:
We will need to use a `rx_entry` for this. tx_entry pool is fixed sized, and cannot grow.